src/System.Security.Claims/src/System/Security/Claims/Claim.cs

```diff
@@ -0,0 +1,550 @@
+                numPropertiesRead++;
+            }
+            else
+            {
+                _originalIssuer = ClaimsIdentity.DefaultIssuer;
+            }
+
+            if ((mask & SerializationMask.HasProperties) == SerializationMask.HasProperties)
+            {
+                // TODO - brentsch, maximum # of properties
+                int numProperties = reader.ReadInt32();
+                for (int i = 0; i < numProperties; i++)
+                {
+                    Properties.Add(reader.ReadString(), reader.ReadString());
+                }
+            }
+
+            if ((mask & SerializationMask.UserData) == SerializationMask.UserData)
+            {
+                // TODO - brentschmaltz - maximum size ??
+                int cb = reader.ReadInt32();
+                _userSerializationData = reader.ReadBytes(cb);
+                numPropertiesRead++;
+            }
+
+            for (int i = numPropertiesRead; i < numPropertiesToRead; i++)
+            {
+                reader.ReadString();
+            }
+        }
+
+        /// <summary>
+        /// Serializes using a <see cref="BinaryWriter"/>
+        /// </summary>
+        /// <param name="writer">the <see cref="BinaryWriter"/> to use for data storage.</param>
+        /// <exception cref="ArgumentNullException">if 'writer' is null.</exception>
+        public virtual void WriteTo(BinaryWriter writer)
+        {
+            WriteTo(writer, null);
+        }
+
+        /// <summary>
+        /// Serializes using a <see cref="BinaryWriter"/>
+        /// </summary>
+        /// <param name="writer">the <see cref="BinaryWriter"/> to use for data storage.</param>
+        /// <param name="userData">additional data provided by derived type.</param>
+        /// <exception cref="ArgumentNullException">if 'writer' is null.</exception>
+        protected virtual void WriteTo(BinaryWriter writer, byte[] userData)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException("writer");
+            }
+
+            // TODO - brentsch, given that there may be a bunch of NameClaims and or RoleClaims, we should account for this.
+            // This would require coordination between ClaimsIdentity and Claim since the ClaimsIdentity defines these 'types'. For now just use defaults.
+
+            int numberOfPropertiesWritten = 1;
+            SerializationMask mask = SerializationMask.None;
+            if (string.Equals(_type, ClaimsIdentity.DefaultNameClaimType))
+            {
+                mask |= SerializationMask.NameClaimType;
+            }
+            else if (string.Equals(_type, ClaimsIdentity.DefaultRoleClaimType))
+            {
+                mask |= SerializationMask.RoleClaimType;
+            }
+            else
+            {
+                numberOfPropertiesWritten++;
+            }
+
+            if (!string.Equals(_valueType, ClaimValueTypes.String, StringComparison.Ordinal))
```

Comment:
Nit: string.Equals(string1, string2) uses ordinal comparison, right?  The above string.Equals calls don't specify a StringComparison, but then this one and the ones below do and specify Ordinal.  Ideally we'd be consistent.